### PR TITLE
Updated firing pattern plot

### PR DESCRIPTION
**Pull request by tcstewar**
Here's the latest version of the "sparkle plot" (a.k.a. "spike grid").  This is a rebased version of #152 with a few additions I made for stability.  Most of the discussion for this PR was in #152 and this PR continues that one.

Two fixes have been added since the discussion there.  First, the time data was being stored completely wrong, so if you tried scrolling backwards in time it just didn't work at all.  This is now fixed.  Second, it now updates the plot when you change the number of neurons.

### CODE REVIEW

**Comment by Seanny123** [comment]
If I resize this plot (chrome/linux-mint) it remains black until I hit the pause button.

**Comment by tcstewar** [comment]
> If I resize this plot (chrome/linux-mint) it remains black until I hit the pause button.

Hmm...  that seems like the intended behaviour to me, as if you haven't hit play yet, then there have been no spikes, and so the plot should be all black.  What do you think it should do?

**Comment by Seanny123** [comment]
I meant that if I moved or resized the plot while the simulation is running, the plot turned black until I hit the pause button. When I hit the pause button, it froze to the last spike data (as it should) and will start again normally once I press play. The undesired behaviour occurs when moving or resizing the plot.

**Comment by tcstewar** [comment]
> I meant that if I moved or resized the plot while the simulation is running, the plot turned black until I hit the pause button. When I hit the pause button, it froze to the last spike data (as it should) and will start again normally once I press play. The undesired behaviour occurs when moving or resizing the plot.

This looks like it's not actually anything to do with resizing, but is to do with timing -- if you're running a very simple model and your simulation gets ahead of the display, then the display goes black (much like the value plots used to disappear).

I've fixed this in the same way as we fixed the value plots -- all the data gets sent at once.  I'm not sure how this interacts with the frame dropping, though, so I'd like to do another round of testing on different machines.

@Seanny123, does this fix it on your machine?

**Comment by Seanny123** [comment]
Yep, that fixed the problem on my machine

File: nengo_gui/components/spike_grid.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                                         self.node, synapse=0.01)
+
+    def remove_nengo_objects(self, page):
+        page.model.connections.remove(self.conn)
+        page.model.nodes.remove(self.node)
+
+    def gather_data(self, t, x):
+        self.max_value = max(self.max_value, np.max(x))
+
+        # TODO: pass only spiking neurons, using subclass of Nengo.Image?
```
Comment (Seanny123) [inline nengo_gui/components/spike_grid.py]: What Nengo.Image are you talking about?

**Comment by Seanny123** [comment]
The code LGTM. After it's merged I can start working on #382

File: nengo_gui/components/spike_grid.py (hunk 1 above)
Comment (tcstewar) [inline nengo_gui/components/spike_grid.py]: Ah, that comment's a bit outdated anyway, as we don't want to just send spike data when we're doing the synapse filter in Nengo.  I removed it.

**Comment by tcstewar** [comment]
> Yep, that fixed the problem on my machine

Cool, and it's worked on the few other machines I've tried, so I'll merge this and see how it goes.

**Pull request merged**